src/dftbp/timedep/timeprop.F90

```diff
@@ -2459,6 +2487,14 @@ subroutine writeTDOutputs(this, dipoleDat, qDat, energyDat, forceDat, coorDat, f
       end if
     end if
 
+    if (this%tWriteAtomEnergies) then
```

Comment:
Can you add a checked example case to the regression test collection? These lines of code are not run with any current input, so are not protected against something breaking on refactoring.

Reply:
I just added a test and tagged output